### DISCUSSION TITLE
Looking for Feedback: Transitioning to ESM

### DISCUSSION THREAD

**Discussion by nicholasgriffintn** [bbc/sqs-consumer#477] [Announcements]
In order to modernise SQS Consumer and be able to use dependencies that have transitioned to ESM correctly, we are looking to transition this module to ESM.

Some of the work has been started here:

https://github.com/bbc/sqs-consumer/pull/476

We are looking for feedback before we officially make this change and would like to know any thoughts that the community might have on this, if they have any at all, feel free to comment if you do :).

Update: This has now been published as the canary version `10.0.0-canary.2`, you can read the release notes for more info: https://github.com/bbc/sqs-consumer/releases/tag/v10.0.0-canary.2

**Comment by nicholasgriffintn** [comment]
In particular, are people generally ready for ESM only? Do we need to complicate the build and support by adding CJS exports too?

**Comment by mdrobny** [comment]
I agree that ESM build should be availabile since that's where the Node.js community is headed (and that's great).

At the same time I would really appreciate CJS build in parallel but that's because many of the applications I work on are still not using ESM because.. they just work 😁

It will of course create a bit of a complexity but I think it's possible to achieve a balance 😉

**Reply by nicholasgriffintn** [reply]
Thanks for the feedback, that's definitely useful information, I was having a think on it and did a bit of research, looks like it may be possible with just `tsc`, which I'm currently investigating.

Would ideally like to avoid a full on build process, we'd need to publish a canary version and test in cjs and esm environments I think.
